src/Dialogs/Device/PortDataField.cpp

```diff
@@ -114,9 +114,11 @@ FillDefaultSerialPorts(DataFieldEnum &df) noexcept
 {
   for (unsigned i = 1; i <= 10; ++i) {
     TCHAR buffer[64];
-    _stprintf(buffer, _T("COM%u:"), i);
-    if (DosDeviceExists(buffer))
-       AddPort(df, DeviceConfig::PortType::SERIAL, buffer);
+    _stprintf(buffer, _T("COM%u"), i);
+    if (DosDeviceExists(buffer)) {
+      _tcscat(buffer, _T(":"));  // add the colon to the path string
```

Comment:
If this colon is not necessary, what's the point of adding overhead here to omit it and then add it? What regression? Your commit message is inadequate.

Reply:
I just can't believe what I'm reading there... In Windows, there can't be a colon in the search string in the port search function '`QueryDosDevice`', but in the port string the one for calling `CreateFile(path,...)` is in the old one Style ('COM1:') still urgently needed! So that the commit first fixes the bug that XCSoar introduced since Windows version 7.24, I removed the colon in the top line, but added it back to AddPort - knowing full well that I won't need this character later. But so that the commit also works as a standalone CherryPick and really fixes the error, that's how I did it! Incidentally, Max removed the '`DosDeviceExists(...)`' function, which he had requested at the time to make the search clearer. Ok, I can live with that - but the fact that it still does NOT find a serial device in the search function under Windows now is very bad! It would be better to show all 9 possible ports like before - regardless of whether they are available or not!But this also makes an extension to port numbers greater than 10 unnecessary, because you can't just add 255 ports to the list...

And apparently Max doesn't even seem to try out the code he's putting in, at least as far as the Windows part is concerned: that was the case more than a year ago, when he added the colon 
to commit [86bc99c](https://github.com/XCSoar/XCSoar/commit/86bc99c847ef31bf237a49f1d7c1a7508ed5ca56) - and it still is still the same today, where the code doesn't work with the current changes either! Max is doing such a great job but even with his limited time budget he has to accept that there are areas that others may have a better grasp of! Especially since this Windows code has no relevance at all for the main functionality of XCSoar on Android, Kobo and OpenVario!